extensions/nyctbustrips/nyctbustrips.go

```diff
@@ -0,0 +1,29 @@
+package nyctbustrips
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/jamespfennell/gtfs/extensions"
+	gtfsrt "github.com/jamespfennell/gtfs/proto"
+)
+
+// Extension returns the NYCT bus trips extension
+func Extension() extensions.Extension {
+	return extension{}
+}
+
+type extension struct {
+	extensions.NoExtensionImpl
+}
+
+func (e extension) UpdateTrip(trip *gtfsrt.TripUpdate, feedCreatedAt uint64) extensions.UpdateTripResult {
+	tripId := trip.Trip.GetTripId()
+	vehicleId := trip.Vehicle.GetId()
```

Comment:
Is it possible that `trip.Trip` or `trip.Vehicle` may be nil here? If so would we get a nil panic?

Perhaps not because the generated proto code for `GetTripId` probably has a nil guard and does the right thing. Even is, it may still be nice to use `trip.GetTrip().GetTripId()` nonetheless?

Reply:
FWIW, looking at the [NYCT trips extension](https://github.com/jamespfennell/gtfs/blob/main/extensions/nycttrips/nycttrips.go) it seems I did use the `GetX` methods everywhere, instead of accessing `X`directly.

Reply:
(Which is not to say that is the right way! Just a data point :) )

Reply:
Good catch, have updated the PR to use `GetX` everywhere.